packages/lab/src/slider/Slider.tsx

```diff
@@ -75,99 +69,59 @@ export const Slider = forwardRef<HTMLDivElement, SliderProps>(function Slider(
     window: targetWindow,
   });
 
-  const trackRef = useRef<HTMLDivElement>(null);
-
   const [value, setValue] = useControlled<SliderValue>({
-    controlled: valueProp,
+    controlled: parseValueProp(valueProp, min, max),
     default: defaultValue,
     name: "Slider",
     state: "Value",
   });
 
-  const updateValueItem = useValueUpdater(pushable, pushDistance, min, max);
-
-  const trackStyle = useMemo(
-    () => createTrackStyle(min, max, value),
-    [min, max, value],
-  );
-
-  const valueLength = Array.isArray(value) ? value.length : 1;
-
-  const handleStyles: CSSProperties[] = useMemo(
-    () => createHandleStyles(valueLength),
-    [valueLength],
-  );
-
-  const onMouseDown = useSliderMouseDown(
-    trackRef,
-    value,
-    min,
-    max,
-    step,
-    updateValueItem,
-    setValue,
-    onChange,
-  );
-
-  const onKeyDown = useSliderKeyDown(
-    value,
-    min,
-    max,
-    pageStep,
-    step,
-    updateValueItem,
-    setValue,
-    onChange,
-  );
+  const handleSliderChange = (value: SliderValue) => {
+    setValue(value);
+    onChange?.(value);
```

Comment:
Is there a reason this can't take an event? All our other change handlers in the library do.